[js] switchTo().window() should use a 'handle' parameter rather than `name`

Standard8 opened the pull request:
- [x] By placing an `X` in the preceding checkbox, I verify that I have signed the [Contributor License Agreement](https://github.com/SeleniumHQ/selenium/blob/master/CONTRIBUTING.md#step-6-sign-the-cla)

This is to match the W3C Webdriver spec, and fixes `switchTo().window()` for use with geckodriver.

Note that the python driver already uses `handle` rather than `name`

I've tried running the tests locally, but either they are taking a very long time, or there's something wrong with them on my machine.

andreastt commented on javascript/firefox-driver/js/nsCommandProcessor.js:
```diff
@@ -533,13 +533,13 @@ nsCommandProcessor.prototype.execute = function(jsonCommandString,
 nsCommandProcessor.prototype.switchToWindow = function(response, parameters,
                                                        opt_searchAttempt) {
-  var lookFor = parameters.name;
+  var lookFor = parameters.handle;
```
/javascript/firefox-driver is the Selenium implementation of a WebDriver for Firefox. Since it generally isn’t W3C compatible, it shouldn’t change.

andreastt commented on javascript/node/selenium-webdriver/lib/webdriver.js:
```diff
@@ -1656,7 +1656,7 @@ class TargetLocator {
     return this.driver_.schedule(
         new command.Command(command.Name.SWITCH_TO_WINDOW).
-            setParameter('name', nameOrHandle),
+            setParameter('handle', nameOrHandle),
```
This should only be sent if the driver is speaking to a W3C conformant remote, so we need an if-condition check like we have in the Python bindings.

Standard8 commented:
@andreastt Thank you for the hints, I've undone the webdriver for Firefox change, and added the W3C compliance detection.

I also worked out how to run the nodejs tests, and I've added a couple of test cases to test that the parameter and w3c detection works correctly.

For reference, cd into `javascript/node/selenium-webdriver`, `npm install` then `npm test`. Need geckodriver in $PATH.

andreastt commented:
r? @jleyba

jleyba commented on javascript/node/selenium-webdriver/lib/webdriver.js:
```diff
@@ -1654,9 +1654,10 @@ class TargetLocator {
    *     when the driver has changed focus to the specified window.
    */
   window(nameOrHandle) {
+    let paramName = this.driver_.getExecutor().w3c ? 'handle' : 'name';
```
I'd rather just send the parameter twice than break encapsulation here. There's already precedence with webelement IDs

jleyba commented:
Looks like we're missing e2e coverage for window switching in

https://github.com/SeleniumHQ/selenium/blob/master/javascript/node/selenium-webdriver/test/window_test.js

Mind adding a test or 2?

Standard8 commented:
> Mind adding a test or 2?

@jleyba It took a bit of work, but I've now added a test, it seems to work fine.
